spec/services/payment_providers/adyen_service_spec.rb

```diff
@@ -58,4 +58,87 @@
       end
     end
   end
+
+  describe '#handle_incoming_webhook' do
+    let(:adyen_provider) { create(:adyen_provider, organization:, hmac_key: nil) }
+
+    let(:body) do
+      JSON.parse(event_response_json)['notificationItems'].first&.dig('NotificationRequestItem')
+    end
+
+    let(:event_response_json) do
+      path = Rails.root.join('spec/fixtures/adyen/webhook_authorisation_response.json')
+      File.read(path)
+    end
+
+    before { adyen_provider }
+
+    it 'checks the webhook' do
+      result = adyen_service.handle_incoming_webhook(
+        organization_id: organization.id,
+        body:
+      )
+
+      expect(result).to be_success
+
+      expect(result.event).to eq(body)
+      expect(PaymentProviders::Adyen::HandleEventJob).to have_been_enqueued
+    end
+
+    context 'when failing to validate the signature' do
+      before do
+        organization.adyen_payment_provider.update! hmac_key: "123"
+      end
+
+      it 'returns an error' do
+        result = adyen_service.handle_incoming_webhook(
+          organization_id: organization.id,
+          body:
+        )
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::ServiceFailure)
+          expect(result.error.code).to eq('webhook_error')
+          expect(result.error.error_message).to eq('Invalid signature')
+        end
+      end
+    end
+  end
+
+  describe '#handle_event' do
+    let(:payment_service) { instance_double(Invoices::Payments::AdyenService) }
+    let(:payment_provider_service) { instance_double(PaymentProviderCustomers::AdyenService) }
+    let(:service_result) { BaseService::Result.new }
+
+    before do
+      allow(Invoices::Payments::AdyenService).to receive(:new)
+        .and_return(payment_service)
+      allow(PaymentProviderCustomers::AdyenService).to receive(:new)
+        .and_return(payment_provider_service)
+      allow(payment_service).to receive(:update_payment_status)
+        .and_return(service_result)
+      allow(payment_provider_service).to receive(:preauthorise)
+        .and_return(service_result)
+    end
+
+    context 'when succeeded authorisation event' do
+      let(:event_json) do
+        JSON.parse(event_response_json)['notificationItems'].
```

Comment:
Layout/DotPosition: Place the . on the next line, together with the method name. (https://rubystyle.guide#consistent-multi-line-chains)